nutkit/frontend/driver.py

```diff
@@ -13,15 +13,12 @@ def __init__(self, backend, uri, auth_token, user_agent=None,
                  max_tx_retry_time_ms=None, encrypted=None,
                  trusted_certificates=None, liveness_check_timeout_ms=None,
                  max_connection_pool_size=None,
-                 connection_acquisition_timeout_ms=None,
-                 bookmark_manager_config: Optional[BMMConfig] = None):
+                 connection_acquisition_timeout_ms=None):
         self._backend = backend
         self._resolver_fn = resolver_fn
         self._domain_name_resolver_fn = domain_name_resolver_fn
-        self._bookmark_manager_config = bookmark_manager_config
-        bookmark_manager = from_bookmark_manager_config_to_protocol(
-            bookmark_manager_config
-        )
+        self._bookmarks_managers = {}
```

Comment:
I kept here because it makes the object more restricted to a scope and it was the more restricted scope I figured out to have it registered and still be able of trigger the extension functions.

The Bookmark Managers still be able to be created as a standalone object, but when associated to the session, the driver object will keep the reference for triggering the callbacks if needed.

About keep the references as global, Driver objects (tk side) accessing the same list of BMM can cause really complicated to debug issues. I prefer not. And the ideal scenario, the Session will hold the reference. However, I don't think it worth the re-factory with the short development time we have.